utils/staking-miner/src/main.rs

```diff
@@ -0,0 +1,523 @@
+		.inject_hashed_prefix(&system_block_hash_key)
+		.build()
+		.await
+		.map_err(|why| Error::RemoteExternalities(why))
+}
+
+/// Compute the election at the given block number. It expects to NOT be `Phase::Off`. In other
+/// words, the snapshot must exists on the given externalities.
+fn mine_unchecked<T: EPM::Config>(
+	ext: &mut Ext,
+	iterations: usize,
+	do_feasibility: bool,
+) -> Result<(EPM::RawSolution<EPM::CompactOf<T>>, u32), Error> {
+	ext.execute_with(|| {
+		let (solution, _) = <EPM::Pallet<T>>::mine_solution(iterations)?;
+		if do_feasibility {
+			let _ = <EPM::Pallet<T>>::feasibility_check(solution.clone(), EPM::ElectionCompute::Signed)?;
+		}
+		let witness = <EPM::SignedSubmissions<T>>::decode_len().unwrap_or_default();
+		Ok((solution, witness as u32))
+	})
+}
+
+#[allow(unused)]
+fn mine_dpos<T: EPM::Config>(
+	ext: &mut Ext,
+) -> Result<(), Error> {
+	ext.execute_with(|| {
+		use EPM::RoundSnapshot;
+		use std::collections::BTreeMap;
+		let RoundSnapshot { voters, .. } = EPM::Snapshot::<T>::get().unwrap();
+		let desired_targets = EPM::DesiredTargets::<T>::get().unwrap();
+		let mut candidates_and_backing = BTreeMap::<T::AccountId, u128>::new();
+		voters.into_iter().for_each(|(who, stake, targets)| {
+			if targets.len() == 0 {
+				println!("target = {:?}", (who, stake, targets));
+				return;
+			}
+			let share: u128 = (stake as u128) / (targets.len() as u128);
+			for target in targets {
+				*candidates_and_backing.entry(target.clone()).or_default() += share
+			}
+		});
+
+		let mut candidates_and_backing =
+			candidates_and_backing.into_iter().collect::<Vec<(_, _)>>();
+		candidates_and_backing.sort_by_key(|(_, total_stake)| *total_stake);
+		let winners = candidates_and_backing
+			.into_iter()
+			.rev()
+			.take(desired_targets as usize)
+			.collect::<Vec<_>>();
+		let score = {
+			let min_staker = *winners.last().map(|(_, stake)| stake).unwrap();
+			let sum_stake = winners.iter().fold(0u128, |acc, (_, stake)| acc + stake);
+			let sum_squared = winners.iter().fold(0u128, |acc, (_, stake)| acc + stake);
+			[min_staker, sum_stake, sum_squared]
+		};
+		println!("mined a dpos-like solution with score = {:?}", score);
+		Ok(())
+	})
+}
+
+pub(crate) async fn get_account_info<T: frame_system::Config>(
+	client: &WsClient,
+	who: &T::AccountId,
+	maybe_at: Option<T::Hash>,
+) -> Result<Option<frame_system::AccountInfo<Index, T::AccountData>>, Error> {
+	rpc_helpers::get_storage::<frame_system::AccountInfo<Index, T::AccountData>>(
+		client,
+		params! {
+			sp_core::storage::StorageKey(<frame_system::Account<T>>::hashed_key_for(&who)),
+			maybe_at
+		},
+	)
+	.await
+}
+
+/// Read the signer account's uri from the given `path`.
+async fn read_signer_uri<
+	P: AsRef<Path>,
+	T: frame_system::Config<AccountId = AccountId, Index = Index>,
+>(
+	path: P,
+	client: &WsClient,
+) -> Result<Signer, Error> {
+	let uri = std::fs::read_to_string(path)?;
+
+	// trim any trailing garbage.
+	let uri = uri.trim_end();
```

Comment:
Maybe not worth it but a micro improvement might be `uri.trim_end().to_string().trim_start()`